src/app/map/[id]/components/controls/layers/BoundariesControl/useBoundariesControl.tsx

```diff
@@ -0,0 +1,178 @@
+import { useCallback, useMemo } from "react";
+import {
+  useDataSources,
+  useMembersDataSource,
+} from "@/app/map/[id]/hooks/useDataSources";
+import { useMapViews } from "@/app/map/[id]/hooks/useMapViews";
+import { MAX_COLUMN_KEY } from "@/constants";
+import { AreaSetGroupCodeLabels } from "@/labels";
+import {
+  CalculationType,
+  ColorScheme,
+  VisualisationType,
+} from "@/server/models/MapView";
+import type { AreaSetGroupCode } from "@/server/models/AreaSet";
+
+export function useBoundariesControl() {
+  const { viewConfig, updateViewConfig } = useMapViews();
+  const membersDataSource = useMembersDataSource();
+  const { data: allDataSources } = useDataSources();
+
+  const hnhVoteShareDataSource = useMemo(
+    () => allDataSources?.find((ds) => ds.name === "HNH Voteshare"),
+    [allDataSources],
+  );
+
+  const isChoroplethVisible = useMemo(
+    () => viewConfig.visualisationType !== VisualisationType.BoundaryOnly,
+    [viewConfig.visualisationType],
+  );
+
+  const toggleChoropleth = useCallback(() => {
+    if (viewConfig.visualisationType === VisualisationType.BoundaryOnly) {
+      updateViewConfig({ visualisationType: VisualisationType.Choropleth });
+    } else {
+      updateViewConfig({ visualisationType: VisualisationType.BoundaryOnly });
+    }
+  }, [viewConfig.visualisationType, updateViewConfig]);
+
+  const fillLabel = useMemo(() => {
+    if (!viewConfig.areaDataSourceId || viewConfig.areaDataSourceId === "") {
+      return "No Fill";
+    }
+
+    if (viewConfig.calculationType === CalculationType.Count) {
+      return "Member Count";
+    }
+
+    if (
+      hnhVoteShareDataSource &&
```

Comment:
request: this should be the 2024 GE Results data source, instead of the Hope Not Hate one